retis/src/collect/collect.rs

```diff
@@ -343,26 +343,38 @@ impl Collectors {
             }
             ok
         };
-        collect
-            .args()?
-            .probes
-            .iter()
-            .try_for_each(|p| -> Result<()> {
-                probe_from_cli(p, filter)?
-                    .drain(..)
-                    .try_for_each(|p| self.probes.builder_mut()?.register_probe(p))?;
-                Ok(())
-            })?;
+        collect.probes.iter().try_for_each(|p| -> Result<()> {
+            probe_from_cli(p, filter)?
+                .drain(..)
+                .try_for_each(|p| self.probes.builder_mut()?.register_probe(p))?;
+            Ok(())
+        })?;
 
         Ok(())
     }
 
     /// Start the event retrieval for all collectors by calling
     /// their `start()` function.
-    pub(crate) fn start(&mut self) -> Result<()> {
+    pub(super) fn start(&mut self, collect: &Collect) -> Result<()> {
         // Create factories.
         #[cfg_attr(test, allow(unused_mut))]
-        let mut section_factories = self.modules.section_factories()?;
+        let mut section_factories = section_factories()?;
+
+        // Configure factories based on collectors config.
+        if let Some(skb_factory) = section_factories.get_mut(&FactoryId::Skb) {
+            skb_factory
+                .as_any_mut()
+                .downcast_mut::<SkbEventFactory>()
+                .ok_or_else(|| anyhow!("Failed to downcast SkbEventFactory"))?
+                .report_eth(
+                    collect
+                        .collector_args
+                        .skb
+                        .skb_sections
+                        .iter()
+                        .any(|s| s == "all" || s == "eth"),
+                );
+        }
```

Comment:
In the lines of my previous comment. We are now opening the door for more of this "custom configurations". Should we let the ~~modules~~ collectors configure their factories here? Something like:
1 - Start the factory _after_ starting collectors (not sure what the implications of this are)
2 - Pass each factory down to it's associated collector (I guess they are not 100% untied)
3 - Each collector can then downcast it and tweak it.

Reply:
I mostly agree. I would just not link a factory to a collector in a 1:1 relationship and just pass the list of factories to all collectors. They can then just downcast the ones they want to tweak.

Reply:
In that case, that would do  it.

Reply:
As discussed, will do as a follow up patch (probably part of the OVS flow enrichment PR)